File: ui/backend/booking-service/src/main/java/com/tripease/booking/dto/RecentBookingsResponseDTO.java
package com.tripease.booking.dto;

import com.tripease.booking.model.BookingStatus;
import com.tripease.booking.model.ResourceType;
import lombok.Builder;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Builder
public record RecentBookingsResponseDTO(
        ResourceType resourceType,
        String subType,
        BookingStatus bookingStatus,
        LocalDate startDate,
        LocalDate endDate,
        BigDecimal totalAmount,
        LocalDateTime createdAt,
        LocalDateTime updatedAt,
        ResourceDetailsDTO resourceDetails
        ) {}

File: ui/frontend/src/app/core/services/dashboard/dto/RecentBookingsResponseDTO.ts
import { ResourceDetailsDTO } from "./ResourceDetailsDTO";


export interface RecentBookingsResponseDTO {
  resourceType: 'HOTEL' | 'FLIGHT'; // Using string literal types for safety
  subType: string;
  bookingStatus: string;
  startDate: string;
  endDate?: string; // Optional for flights
  totalAmount: number;
  createdAt: string;
  updatedAt: string;
  resourceDetails: ResourceDetailsDTO;
}